mobility/choreographer/src/choreographer_nodelet.cc

```diff
@@ -291,6 +291,7 @@ class ChoreographerNodelet : public ff_util::FreeFlyerNodelet {
         }
         states_.insert(states_.begin(), current_pose);
         replan_attempts_++;
+        ros::Duration(1).sleep();  // sleep for a second
```

Comment:
This is necessary in between planning attempts in case the obstacle map has some old obstacles that are not valid anymore, it allows time to update the map

Reply:
This goes against the spirit of the finite state machine, which should have very quick functions in this part... with this sleep, the state machine can't be interrupted for example with a cancel event.

If we are going to do this, we should have a separate sleeping state and have an event trigger the FSM when a timer goes off.

However, I think this is a very special case where we expect the cargo to move. I kind of think instead of putting this wait in the choreographer it might make sense to try multiple times at the behavior level. In the general case, when we don't expect things to move, it would be better to fail quickly. Or we could make it so we can change the wait time and retries when we call the choreographer--- maybe that is the best option, but I'm not sure what changes it would take. What do you think?

Reply:
This is a problem that actually shows up in all qp planner trajectories and not specific to the cargo demo. When the trajectory detects obstacles and fails, sometimes when the obstacle map is updated because of the "memory_time" function, the objects previously detected are updated and the robot is able to continue instead of getting stuck. This delay allows the mapper to get rid of older obstacles that are not in the fov of the robot.

Given this, we should probably go with approach 1?